### PR TITLE
Ignore `cursorOffset` in `format()` and `check()`

### PR DESCRIPTION
**Pull request by fisker**
## Description

Maintain `cursorOffset` seems expensive to me, and it's useless in `format()` and `check()`

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -55,14 +55,16 @@ function withPlugins(
 
 const formatWithCursor = withPlugins(core.formatWithCursor);
 
-async function format(text, opts) {
-  const { formatted } = await formatWithCursor(text, opts);
+async function format(text, options) {
+  const { formatted } = await formatWithCursor(text, {
+    ...options,
```
Comment (sosukesuzuki) [inline src/index.js:60]: I don't think we should use spread syntax for this because it is little expensive. Can we use `options.cursorOffset = -1`?

Comment (fisker) [reply src/index.js:60]: I don't think so, this option is user input, we shouldn't modify it. It can also be `undefined`.

Comment (fisker) [reply src/index.js:60]: I don't think clone an object is that expensive.

Comment (sosukesuzuki) [reply src/index.js:60]: Okay, we can keep it.

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**